src/Lykke.Common.Validation/ImageType/ImageTypeFluentValidator.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using FluentValidation.Validators;
+using Microsoft.AspNetCore.Http;
+
+namespace Lykke.Common.Validation.ImageType
+{
+    /// <summary>
+    ///     Deep image validator.
+    ///     Validates extension and file signature by checking first bytes.
+    ///     Available file extensions: ".jpg", ".jpeg", ".png", ".gif", ".bmp".
+    ///     Any other extensions would be ignored. 
+    /// </summary>
+    public class ImageTypeFluentValidator : PropertyValidator
+    {
+        private readonly IDictionary<string, ImageType> _imageTypes = new Dictionary<string, ImageType>();
+
+        public ImageTypeFluentValidator(params ImageType[] imageTypes) : base(
+            "{PropertyName} must be an image file.")
```

Comment:
It's possibly better to include validation rules into validation error.

I.e. {PropertyName} must be an image in following formats { ".jpg", ".jpeg", ".gif" }

Reply:
Done